server/bin/pbench-backup-tarballs.py

```diff
@@ -354,27 +353,9 @@ def backup_data(s3_obj, config, logger):
 
         if local_backup_result == Status.SUCCESS \
             and s3_backup_result == Status.SUCCESS:
-            # moved to backed-up
-            _rename_tb_link(tb, os.path.join(
-                controller_path, "BACKED-UP"), logger)
-        elif local_backup_result == Status.SUCCESS \
-            and s3_backup_result == Status.FAIL:
-            # move to backed-up-local
-            _rename_tb_link(tb, os.path.join(
-                controller_path, "BACKED-UP-LOCAL"), logger)
-        elif local_backup_result == Status.FAIL \
-            and s3_backup_result == Status.SUCCESS:
-            # move to backed-up-S3
-            _rename_tb_link(tb, os.path.join(
-                controller_path, "BACKED-UP-S3"), logger)
-        elif local_backup_result == Status.FAIL \
-            and s3_backup_result == Status.FAIL:
-            # move tp backed-up-failed
+            # Move tar ball symlink to its final resting place
             _rename_tb_link(tb, os.path.join(
-                controller_path, "BACKED-UP-FAILED"), logger)
-        else:
-            assert False, "Logic bomb: cannot happen: local_backup_result = \
-                {}, s3_backup_result = {}".format(local_backup_result,                                           s3_backup_result)
+                controller_path, _linkdest), logger)
```

Comment:
So if one of the backup methods fails, we are going to keep retrying?  I don't think that's a good idea: chances are that the failure is not transient, so we are going to keep failing once a minute until somebody notices.

Reply:
Try to avoid transient backup errors from preventing a tar ball from being backed up at all.

If we have a transient error, in the current scheme the failure will move the symlink out of TO-BACKUP and it will never be backed up until somebody takes manual action to do something about.

In this proposed scheme, the backup process will keep re-trying, so when the transient error is cleared, the backup will succeed.

If there is a better way to approach this, feel free!